vyper/context/validation/local.py

```diff
@@ -350,9 +369,20 @@ def visit_Expr(self, node):
             raise StructureException("To call an event you must use the `log` statement", node)
 
         if isinstance(fn_type, ContractFunctionType):
-            if self.func.is_constant and not fn_type.is_constant:
+            if self.func.mutability is StateMutability.VIEW and fn_type.mutability not in (
+                StateMutability.VIEW,
+                StateMutability.PURE,
+            ):
```

Comment:
It would be nicer to do `self.func.mutability is StateMutability.VIEW and fn_type.mutability > StateMutability.VIEW`

or better yet: `self.func.mutability < fn_type.mutability`, and combine with below

Reply:
Actually, is it okay that a `nonpayable` function can call a `payable` one?

Reply:
Yeah, nonpayable should be able to call payable.  It doesn't mean "can't transfer ETH", just "can't receive ETH".